app/Actions/Sharing/ListShare.php

```diff
@@ -72,7 +73,7 @@ public function do(int $userId): Shares
 		}
 	}
 
-	private function breadcrumbPath(object $album): string
+	private function breadcrumbPath(Album $album): string
```

Comment:
I am not sure whether I like this change. The passed object is not an album, but the result of a specially hand-crated DB query. In particular, the object missed most of the properties of an album. Hence, it is wrong to declare it as such.

Reply:
True.
But how would you solve that one ? Inner class ? Because otherwise PhpStan is complaining about unknown properties on Object. :/

Reply:
I didn't know that PHP supported inner classes. We are talking about something like

class OuterClass {
  public class InnterClass {
    ...
  }
}

Right? If not, a DTO could also help.